megatron/data/blendable_dataset.py

```diff
@@ -13,16 +13,39 @@
 from megatron import print_rank_0
 from megatron.core import mpu
 
-class BlendableDataset(torch.utils.data.Dataset):
-
 
-    def __init__(self, datasets, weights, size, *,
-                 data_cache_path=None):
+class BlendableDataset(torch.utils.data.Dataset):
 
+    def __init__(
+            self,
+            datasets,
+            weights,
+            size,
+            *,
+            data_cache_path=None
+    ):
         self.datasets = datasets
         num_datasets = len(datasets)
+        ndsets = np.array([num_datasets], dtype=np.int64).item()
+        len_weights = np.array([len(weights)], dtype=np.int64).item()
+        print_rank_0(f'{len(datasets)=}')
+        print_rank_0(f'{len(weights)}')
+        print_rank_0(f'{ndsets=}')
+        print_rank_0(f'{len_weights=}')
+        # if int(num_datasets) != int(len(weights)):
+        # if f'{num_datasets}' == f'{len(weights)}':
+        if ndsets != len_weights:
+            warr = np.array(weights)
+            print_rank_0('\n'.join([
+                f'{num_datasets=}',
+                f'{len(weights)=}',
+                f'{warr.shape=}',
+                f'{warr.sum()=}',
+                # f'{num_datasets=} != {len(warr)=}',
+            ]))
```

Comment:
Can we merge these prints into single line and add description to make it understandable? Also please remove the commented code (applies to other places as well).